bayesian/utility/v2/stack_analyses.py

```diff
@@ -0,0 +1,243 @@
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # Below ecosystem map tries to find the map entry for given ecosystem.
+        ecosystem_map = {
+            "node": "npm",
+            "python": "pypi",
+            "java": "maven"
+        }
+        # If given ecosystem is not found in above map, than uses the value passed in the request.
+        self.ecosystem = ecosystem_map.get(self.ecosystem, self.ecosystem)
+        current_app.logger.info("Final ecosystem after mapping is '{}'".format(self.ecosystem))
+
+        # Ecosystem should be a valid value
+        if not check_for_accepted_ecosystem(self.ecosystem):
+            error_message = "Error processing request. " \
+                            "'{}' ecosystem is not supported".format(self.ecosystem)
+            current_app.logger.error(error_message)
+            raise HTTPError(400, error=error_message)
+
+    def prepare_request(self):
+        """Prepare post request by creating manifest file info and unique request id."""
+        # Build manifest info from manifest file and path. It read content in utf-8 encoding.
+        self._manifest_file_info = {
+            "filename": self.manifest.filename,
+            "filepath": self.file_path,
+            "content": self.manifest.read().decode('utf-8')
+        }
+        current_app.logger.info(self._manifest_file_info)
+
+        # Generate unique request id using UUID, also record timestamp in readable form
+        self._new_request_id = str(uuid.uuid4().hex)
+        self._date_str = str(datetime.datetime.now())
+
+    def get_request_data(self, external_new_request_id):
+        """Read request data for given request id from RDS."""
+        db_result = fetch_sa_request(rdb, external_new_request_id)
+        if db_result is None:
+            error_message = "Invalid request ID '{}'.".format(external_new_request_id)
+            current_app.logger.error(error_message)
+            raise HTTPError(404, error=error_message)
+
+        return db_result
+
+    def get_stack_result(self, external_new_request_id):
+        """Read and return stack result from RDS."""
+        return retrieve_worker_result(rdb, external_new_request_id, "stack_aggregator_v2")
+
+    def get_recommendation_data(self, external_new_request_id):
+        """Read and return recommendation data from RDS."""
+        return retrieve_worker_result(rdb, external_new_request_id, "recommendation_v2")
+
+    def _read_deps_and_packages(self):
+        """Read dependencies and packages information from manifest file content."""
+        deps = {}
+        packages = []
+
+        # Dependency finder
+        d = DependencyFinder()
+        deps = d.scan_and_find_dependencies(self.ecosystem, [self._manifest_file_info],
+                                            self.show_transitive)
+
+        # Build package details.
+        for p in deps['result'][0]['details'][0]['_resolved']:
```

Comment:
Can we be a little more defensive here? There are a lot of objects that could probably be None.

Reply:
Sure, will add extra safety net.

Reply:
This is fixed in commit 'Review comment to HTTPError from base classes is fixed.'